### PR TITLE
feat(CLI): worker and cpu optimization

### PR DESCRIPTION
**Pull request by ShiningRush**
Optimize the configuration of APISIX in the container environment. 
For more details, please refer to this [article](https://github.com/ShiningRush/memos/tree/master/openresty-note/cpu-setting-in-cloudnative)

@membphis @moonming 
The current default configuration is the best configuration on the physical machine, but this configuration is not good for container.
Should our default configuration be adapted to the container environment or the physical machine first?

### CODE REVIEW

File: bin/apisix [hunk 1]
```diff
@@ -821,6 +821,11 @@ local function init()
         sys_conf["dns_resolver"] = dns_addrs
     end
 
+    local env_worker_processes = os.getenv("AS_WORKER_PROCESSES")
```
Comment (moonming) [inline bin/apisix:824]: how about change `AS_` prefix to `APISIX_`? which is more clear

Comment (ShiningRush) [reply bin/apisix:824]: How about `APIX_`?
The prefix generally does not exceed four letters, such as `kubernetes -> kube`, `prometheus -> prom`, `grafana-> gf`

**Comment by moonming** [comment]
OK

Vinci Xu <notifications@github.com> 于 2020年9月3日周四 下午9:02写道：

**Review by moonming — APPROVED** [review]

**Review by membphis — COMMENTED** [review]
missing test case, please take a look at:

https://github.com/apache/apisix/blob/master/.travis/apisix_cli_test.sh

**Comment by spacewander** [comment]
Ping @ShiningRush ?

**Comment by ShiningRush** [comment]
I'm sorry to forget to fix this PR, I will deal with it this week

**Comment by ShiningRush** [comment]
@YuanYingdong @spacewander fixed, plz have a look.

File: .travis/apisix_cli_test.sh [hunk 2]
```diff
@@ -408,3 +408,36 @@ if ! grep "lua_shared_dict my_dict 1m;" conf/nginx.conf > /dev/null; then
...
+
+make init
+
+count=`grep -c "worker_processes 8;" conf/nginx.conf || true`
+if [ $count -ne 1 ]; then
+    echo "failed: worker_processes is not 8 when using env to set worker processes"
+    exit 1
+fi
+
+echo "passed: worker_processes is not 8 when using env to set worker processes"
```
Comment (membphis) [inline .travis/apisix_cli_test.sh:443]: bad description

**Review by membphis — APPROVED** [review]

File: .travis/apisix_cli_test.sh [hunk 3]
```diff
@@ -408,3 +408,36 @@ if ! grep "lua_shared_dict my_dict 1m;" conf/nginx.conf > /dev/null; then
...
+echo '
+nginx_config:
+  enable_cpu_affinity: false
+' > conf/config.yaml
+
+make init
+
+count=`grep -c "worker_cpu_affinity" conf/nginx.conf  || true`
+if [ $count -ne 0 ]; then
+    echo "failed: nginx.conf file disable cpu affinity"
```
Comment (spacewander) [inline .travis/apisix_cli_test.sh:424]: Should be `enable cpu ...`?

**Review by spacewander — CHANGES_REQUESTED** [review]

File: .travis/apisix_cli_test.sh (hunk 3 above)
Comment (ShiningRush) [reply .travis/apisix_cli_test.sh:424]: This case is used to disable cpu affinity.

Comment (spacewander) [reply .travis/apisix_cli_test.sh:424]: @ShiningRush 
My word is not clear. Actually I mean the error message should show why the test is failed, for example, "failed: nginx.conf file should disable cpu affinity but it doesn't".

Comment (ShiningRush) [reply .travis/apisix_cli_test.sh]: Got it.

**Review by spacewander — APPROVED** [review]

**Pull request merged**